main.c

```diff
@@ -218,6 +231,54 @@ static const struct wl_keyboard_listener keyboard_listener = {
 	.modifiers = noop,
 };
 
+static void touch_handle_down(void *data, struct wl_touch *touch, uint32_t serial,
+                          uint32_t time, struct wl_surface *surface, int32_t id,
+                          wl_fixed_t x, wl_fixed_t y) {
+	struct slurp_seat *seat = data;
+	if (!seat->already_touched) {
```

Comment:
Instead of having a `bool already_touched` field, can we store the touch `id` and check it's the correct one in `touch_handle_motion`?

Reply:
Fixed 585646d2ad2a322da034b52fa494d697584ed144. Turns out this also fixes segfaults when attempting to select with different fingers, thanks for this one.